compiler/noirc_evaluator/src/ssa/parser/tests.rs

```diff
@@ -139,6 +139,18 @@ fn test_multiple_blocks_and_jmp() {
 
 #[test]
 fn test_jmpif() {
+    let src = "
+        acir(inline) fn main f0 {
+          b0(v0: Field):
+            jmpif v0 then: b1, else: b2
+          b1():
+            return
+          b2():
+            return
+        }
+        ";
+    assert_ssa_roundtrip(src);
```

Comment:
Please add a more complex `assert_ssa_roundtrip` along these lines

Reply:
What kind of test were you thinking? In a way we do have many SSA parser tests through our SSA pass tests which utilize the parser.

Reply:
Because this involves the ordering of jumps, I was thinking of a program that jumps in a few different "directions" to ensure the ordering is correct, e.g.

```
b -> c, d
foo -> a, b
c -> e, d
d, e -> return
a -> c, b
```

where `A -> B, C` is expanded to `fn A(x) { jmpif x then B(x) else C(x)`.

Because most of our SSA tests define programs in a dependency-respecting order, e.g.

```
a -> b, c
b -> c, d
..
```

Reply:
> where A -> B, C is expanded to fn A(x) { jmpif x then B(x) else C(x)

I am a bit confused by this notation as it diverges from our SSA and looks to be mixing functions and blocks. Either way I'll add another more complex test.

Reply:
Something like:

```noir
        acir(inline) fn main f0 {
          b0(v0: Field):
            v1 = call f1(v0) -> Field
            v2 = call f1(v0) -> Field
            jmpif v0 then: b0; else: b1;
          b1(v0: Field):
            return v1
          b0(v0: Field):
            return v2
        }
        ..
```

Reply:
I added a test with two jmpifs and 3 different blocks we may jump to